From 012fe5c9bab6ede2507699ffd3603d9eea92d469 Mon Sep 17 00:00:00 2001
From: rump <drupal@seandunaway.com>
Date: Wed, 28 Sep 2011 13:35:43 -0700
Subject: [PATCH] Issue #1023384 by rump: Moving ga.js to <head>.

---
 googleanalytics.module |   32 +++++++++++++-------------------
 1 files changed, 13 insertions(+), 19 deletions(-)

diff --git a/googleanalytics.module b/googleanalytics.module
index 7ccccb6..b765955 100644
--- a/googleanalytics.module
+++ b/googleanalytics.module
@@ -36,6 +36,15 @@ function googleanalytics_theme() {
 }
 
 /**
+ * Implementation of hook_theme_registry_alter().
+ */
+function googleanalytics_theme_registry_alter(&$registry) {
+  // Add custom preprocess function to the beginning of the stack.
+  // This allows use of the drupal_add_js() function before $vars is populated.
+  array_unshift($registry['page']['preprocess functions'], 'googleanalytics_add_js');
+}
+
+/**
  * Implementation of hook_perm().
  */
 function googleanalytics_perm() {
@@ -60,9 +69,9 @@ function googleanalytics_menu() {
 }
 
 /**
- * Implementation of hook_init().
+ * Adds Google Analytics tracking scripts.
  */
-function googleanalytics_init() {
+function googleanalytics_add_js(&$vars, $hook) {
   global $user;
 
   $id = variable_get('googleanalytics_account', '');
@@ -93,21 +102,6 @@ function googleanalytics_init() {
       drupal_add_js(array('googleanalytics' => $link_settings), 'setting', 'header');
       drupal_add_js(drupal_get_path('module', 'googleanalytics') .'/googleanalytics.js', 'module', 'header');
     }
-  }
-}
-
-/**
- * Implementation of hook_footer() to insert JavaScript at the end of the page.
- */
-function googleanalytics_footer($main = 0) {
-  global $user;
-
-  $id = variable_get('googleanalytics_account', '');
-
-  // 1. Check if the GA account number has a value.
-  // 2. Track page views based on visibility value.
-  // 3. Check if we should track the currently active user's role.
-  if (!empty($id) && _googleanalytics_visibility_pages() && _googleanalytics_visibility_user($user)) {
 
     // Add User profile segmentation values.
     $profile_fields = variable_get('googleanalytics_segmentation', array());
@@ -268,7 +262,7 @@ function googleanalytics_footer($main = 0) {
     $script .= 's.parentNode.insertBefore(ga, s);';
     $script .= '})();';
 
-    drupal_add_js($script, 'inline', 'footer');
+    drupal_add_js($script, 'inline', 'header');
   }
 }
 
@@ -491,7 +485,7 @@ function _googleanalytics_visibility_roles($account) {
 function _googleanalytics_visibility_pages() {
   static $page_match;
 
-  // Cache visibility setting in hook_init for hook_footer.
+  // Cache visibility setting in googleanalytics_add_js.
   if (!isset($page_match)) {
 
     $visibility = variable_get('googleanalytics_visibility', 0);
-- 
1.7.5.4

